virtual_rainforest/models/hydrology/hydrology_model.py

```diff
@@ -571,92 +631,114 @@ def calculate_vertical_flow(
     :math:`q` in :math:`\frac{m^3}{s^1}` considering the effective hydraulic
     conductivity:
 
-    :math:`q = K(S)*A*\frac{dh}{dl}`
+    :math:`q = - K(S)*(\frac{dh}{dl}-1)`
 
-    where :math:`A` is the column cross section area (which will be dropped in the
-    conversion to mm) :math:`\frac{dh}{dl}` is the hydraulic gradient with :math:`l` the
-    length of the flow path in meters (here equal to the soil depth). We assume the
-    whole soil column as one layer and the soil moisture to be the mean soil moisture.
+    where :math:`\frac{dh}{dl}` is the hydraulic gradient with :math:`l` the
+    length of the flow path in meters (here equal to the soil depth).
 
     Note that there are severe limitations to this approach on the temporal and
     spatial scale of this model and this can only be treated as a very rough
-    approximation! Further, we do not remove the water from the soil but assume a
-    steady state. To consider the latter, this function should be replaced with a more
-    sophisticaed, multi-layer model or a simple residence time assumption.
+    approximation!
 
     Args:
-        soil_moisture: (mean) Volumetric relative water content in top soil, [unitless]
-        soil_depths: soil depths = length of the flow path, [m]
+        soil_moisture: Volumetric relative water content in top soil, [unitless]
+        soil_layer_thickness: Thickness of all soil_layers, [mm]
         soil_moisture_capacity: soil moisture capacity, [unitless]
         soil_moisture_residual: residual soil moisture, [unitless]
         hydraulic_conductivity: hydraulic conductivity of soil, [m/s]
         hydraulic_gradient: hydraulic gradient (change in hydraulic head) along the flow
             path, positive values indicate downward flow, [m/m]
-        timestep_conversion_factor: factor to convert flow from m^3 per second to mm per
-            model time step
         nonlinearily_parameter: dimensionless parameter in van Genuchten model that
             describes the degree of nonlinearity of the relationship between the
             volumetric water content and the soil matric potential.
+        groundwater_capacity: storage capacity of groupwater
+        seconds_to_day: factor to convert between second and day
 
     Returns:
         volumetric flow rate of water, [mm/timestep]
     """
     m = 1 - 1 / nonlinearily_parameter
```

Comment:
These parameters are everywhere just called `n` and `m=1-1/n`, I'll call it `shape_parameter` which is not completely wrong I guess

Reply:
Yeah - `pyrealm` is absolutely full of `xi` and `phi` and `mj` etc, so I'm being kinda unfair here. I guess it sticks out here because we are using long informative variable names so much. I think that is worthwhile because we hope people will be using our models as the basis for new implementations.

Reply:
I agree 👍

Reply:
I absolutely hate reading code that is so cryptic `tt=st[i] + x /l `, like what???